src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -270,6 +270,36 @@ public new bool IsSectionDefined(string name)
             return zone != null && zone.Items.Count > 0;
         }
 
+        public new void DefineSection(string name, Microsoft.AspNetCore.Mvc.Razor.RenderAsyncDelegate section)
+        {
+            var zone = ThemeLayout.Zones[name];
+            dynamic sectionShape = New.AspSection(SectionContent: SectionDelegate(name,section) ).GetAwaiter().GetResult();
```

Comment:
Also here, `GetAwaiter().GetResult()` on something that could be frequently called and time consuming

Reply:
Add a space after the comma, no space before a closing parenthesis, please use ctrl K+D under the VStudio editor ;)

Reply:
> Also here, `GetAwaiter().GetResult()` on something that could be frequently called and time consuming

I don't have a choice, ShapeFactory has asyc method but compiler doesn't like asyc `DefineSection`